llvm/lib/Target/NVPTX/NVPTXISelLowering.cpp

```diff
@@ -5614,17 +5614,98 @@ static SDValue TryMULWIDECombine(SDNode *N,
   return DCI.DAG.getNode(Opc, DL, MulType, TruncLHS, TruncRHS);
 }
 
+static SDValue matchMADConstOnePattern(SDValue X, SDValue Add) {
+  if (Add->getOpcode() != ISD::ADD)
+    return SDValue();
+
+  SDValue Y = Add->getOperand(0);
+  ConstantSDNode *Const = dyn_cast<ConstantSDNode>(Add->getOperand(1));
```

Comment:
Are we guaranteed to have const operand to be last? I think we normalize them, but I'm not 100% sure it's always the case.

Reply:
Good point, I've added the other case as well just in case.